libvast/include/vast/concept/printable/vast/json.hpp

```diff
@@ -10,21 +10,63 @@
 
 #include "vast/concept/printable/core/operators.hpp"
 #include "vast/concept/printable/core/printer.hpp"
-#include "vast/concept/printable/core/sequence.hpp"
-#include "vast/concept/printable/std/chrono.hpp"
+#include "vast/concept/printable/to_string.hpp"
 #include "vast/data.hpp"
-#include "vast/detail/escapers.hpp"
+#include "vast/detail/string.hpp"
 #include "vast/view.hpp"
 
+#include <fmt/color.h>
 #include <fmt/format.h>
 
+#include <optional>
+
 namespace vast {
 
+struct json_style {
+  fmt::text_style null_;
+  fmt::text_style false_;
+  fmt::text_style true_;
+  fmt::text_style number;
+  fmt::text_style string;
+  fmt::text_style array;
+  fmt::text_style object;
+  fmt::text_style field;
+  fmt::text_style comma;
+};
+
+// Defined in
+// https://github.com/jqlang/jq/blob/c99981c5b2e7e7d4d6d1463cf564bb99e9f18ed9/src/jv_print.c#L27
+inline auto jq_style() -> json_style {
+  return {
+    .null_ = fmt::emphasis::bold | fg(fmt::terminal_color::black),
+    .false_ = fg(fmt::terminal_color::white),
+    .true_ = fg(fmt::terminal_color::white),
+    .number = fg(fmt::terminal_color::white),
+    .string = fg(fmt::terminal_color::green),
+    .array = fmt::emphasis::bold | fg(fmt::terminal_color::white),
+    .object = fmt::emphasis::bold | fg(fmt::terminal_color::white),
+    .field = fmt::emphasis::bold | fg(fmt::terminal_color::blue),
+    .comma = fmt::emphasis::bold | fg(fmt::terminal_color::white),
+  };
+}
+
+inline auto no_style() -> json_style {
+  return {};
+}
+
+inline auto default_style() -> json_style {
+  // TODO: perform TTY detection here.
+  return no_style();
+}
```

Comment:
We discussed in Slack that this cannot feasibly be done in the printer, but rather needs to be done by the saver. Can you remove this, and write a follow-up story for evaluating our options for this?

Reply:
@Dakostu takes the PR from here. I'll let him address this at his convenience.

Reply:
I wrote an issue here:
https://github.com/tenzir/issues/issues/646

I will change that TODO.

Reply:
https://github.com/tenzir/tenzir/pull/3343/commits/a97747f6c6843df9dbf0a4778a03322baad7b6a4